CIP-0033/README.md

```diff
@@ -0,0 +1,117 @@
+---
+CIP: 33
+Title: Reference scripts
+Authors: Michael Peyton Jones <michael.peyton-jones@iohk.io>
+Comments-Summary: No comments
+Comments-URI: 
+Status: Draft
+Type: Standards Track
+Created: 2021-11-29
+License: CC-BY-4.0
+Requires: CIP-31
+---
+
+# Reference scripts
+
+## Abstract
+
+We propose to allow scripts ("reference scripts") to be attached to outputs, and to allow reference scripts to be used to satisfy script requirements during validation, rather than requiring the spending transaction to do so.
+This will allow transactions using common scripts to be much smaller.
+
+## Motivation
+
+Script sizes pose a significant problem. This manifests itself in two ways:
+1. Every time a script is used, the transaction which caused the usage must supply the whole script as part of the transaction. This bloats the chain, and passes on the cost of that bloat to users in the form of transaction size fees.
+2. Transaction size limits become are for users. Even if individual scripts do not hit the limits, using multiple scripts has a proportionally greater risk of hitting the limit.
```

Comment:
I'm not sure what you meant to write here.

Reply:
Thanks, missing a word.